EdgeTagSDKTests/EdgeTagSDKTests.swift

```diff
@@ -0,0 +1,274 @@
+            if success{
+                edgeTagManager.addUserIDGraph(userKey: "email", userValue: "me@domain.com", completion: { success, error in
+                    XCTAssert(success)
+                    exp.fulfill()
+                })
+            }
+        })
+        waitForExpectations(timeout: 40, handler: nil)
+    }
+    
+    func testGetData()
+    {
+        let edgeConfiguration = EdgeTagConfiguration(withUrl: "https://sdk-demo-t.edgetag.io",shouldFetchIDFA: true,disableConsentCheck: false)
+        let edgeTagManager = EdgeTagManager.shared
+        let exp = expectation(description: "\(#function)\(#line)")
+        
+        edgeTagManager.initEdgeTag(withEdgeTagConfiguration: edgeConfiguration, completion: { success, error in
+            if success{
+                edgeTagManager.getDataIDGraph(idGraphKeys: ["cutomInfo","numberValue","testBool","email","invalid value"], completion: { success, error, idGraph in
+                    XCTAssert(success)
+                    exp.fulfill()
+                })
+            }
+        })
+        waitForExpectations(timeout: 40, handler: nil)
+    }
+    
+    func testGetKeys()
+    {
+        let edgeConfiguration = EdgeTagConfiguration(withUrl: "https://sdk-demo-t.edgetag.io",shouldFetchIDFA: true,disableConsentCheck: false)
+        let edgeTagManager = EdgeTagManager.shared
+        let exp = expectation(description: "\(#function)\(#line)")
+        
+        edgeTagManager.initEdgeTag(withEdgeTagConfiguration: edgeConfiguration, completion: { success, error in
+            if success{
+                edgeTagManager.getUserKeys(completion: { success, error, idGraphKeys in
+                    XCTAssert(success)
+                    exp.fulfill()
+                })
+            }
+        })
+        waitForExpectations(timeout: 40, handler: nil)
+    }
+    
+    
+    func testCreateKVForUserData(){
+        PackageProviders.shared.createKVForUserData(kvUserData: ["email":"user@testMail.com"])
+    }
+    
+    func testGetKVForUserData(){
+        let dict = PackageProviders.shared.getKVForUserData()
+        XCTAssert(dict.keys.count > 0)
+    }
+    
+    func testpassIDFAValuesToManager(){
+        IDFAHandler.shared.passIDFAValuesToManager(checkForIDFA: true, idfaAccessGranted: true)
+        XCTAssert(NetworkManager.shared.idfaAccessGranted)
+    }
+    
+    func testSendAppInstallEvent(){
+        StorageHandler.shared.saveAppInstallEventSent()
+        XCTAssert(StorageHandler.shared.getAppInstallEventSent())
+    }
+    
+    func testGetCookie(){
+        
+        let edgeConfiguration = EdgeTagConfiguration(withUrl: "https://sdk-demo-t.edgetag.io",shouldFetchIDFA: true,disableConsentCheck: false)
+        let edgeTagManager = EdgeTagManager.shared
+        
+        edgeTagManager.initEdgeTag(withEdgeTagConfiguration: edgeConfiguration, completion: { success, error in
+            if success{
+                let cookieStr = StorageHandler.shared.getCookie()
```

Comment:
StorageHandler  exposed ? or its available with in the SDK packages ?

Reply:
For test cases the entire SDK code is available to the class using '@testable import'